mysql-test/suite/galera/t/MDEV-30732.test

```diff
@@ -0,0 +1,212 @@
+UPDATE t SET c2=NULL WHERE c = 1;
+UPDATE t SET c2="" WHERE c=1;
+UPDATE t SET c2="TESTtest" WHERE c = 1;
+UPDATE t SET c2="BUGbug" WHERE c2 = "TESTtest";
+DELETE FROM t WHERE c2="BUGbug";
+SELECT * FROM t;
+DROP TABLE t;
+CREATE TABLE t (c INT NOT NULL PRIMARY KEY,c2 VARCHAR(270) NOT NULL,KEY k2 (c2 (6),c)) DEFAULT CHARSET=latin1 ROW_FORMAT=COMPACT;
+INSERT INTO t (c) VALUES (1),(2),(3),(4),(5);
+UPDATE t SET c2=NULL WHERE c = 1;
+UPDATE t SET c2="" WHERE c=1;
+UPDATE t SET c2="TESTtest" WHERE c = 1;
+UPDATE t SET c2="BUGbug" WHERE c2 = "TESTtest";
+DELETE FROM t WHERE c2="BUGbug";
+SELECT * FROM t;
+DROP TABLE t;
+CREATE TABLE t (c INT,c2 CHAR(80),KEY k2 (c2 (6),c)) DEFAULT CHARSET=latin1 ROW_FORMAT=COMPACT;
+INSERT INTO t (c) VALUES (1),(1),(1),(1),(1);
+UPDATE t SET c2=NULL WHERE c = 1;
+UPDATE t SET c2="" WHERE c=1;
+UPDATE t SET c2="TESTtest" WHERE c = 1;
+UPDATE t SET c2="BUGbug" WHERE c2 = "TESTtest";
+DELETE FROM t WHERE c2="BUGbug";
+SELECT * FROM t;
+DROP TABLE t;
+CREATE TABLE t (c INT,c2 CHAR(80) NOT NULL,KEY k2 (c2 (6),c)) DEFAULT CHARSET=latin1 ROW_FORMAT=COMPACT;
+INSERT INTO t (c) VALUES (1),(1),(1),(1),(1);
+UPDATE t SET c2=NULL WHERE c = 1;
+UPDATE t SET c2="" WHERE c=1;
+UPDATE t SET c2="TESTtest" WHERE c = 1;
+UPDATE t SET c2="BUGbug" WHERE c2 = "TESTtest";
+DELETE FROM t WHERE c2="BUGbug";
+SELECT * FROM t;
+DROP TABLE t;
+CREATE TABLE t (c INT NOT NULL PRIMARY KEY,c2 CHAR(80),KEY k2 (c2 (6),c)) DEFAULT CHARSET=latin1 ROW_FORMAT=COMPACT;
+INSERT INTO t (c) VALUES (1),(2),(3),(4),(5);
+UPDATE t SET c2=NULL WHERE c = 1;
+UPDATE t SET c2="" WHERE c=1;
+UPDATE t SET c2="TESTtest" WHERE c = 1;
+UPDATE t SET c2="BUGbug" WHERE c2 = "TESTtest";
+DELETE FROM t WHERE c2="BUGbug";
+SELECT * FROM t;
+DROP TABLE t;
+CREATE TABLE t (c INT NOT NULL PRIMARY KEY,c2 CHAR(80) NOT NULL,KEY k2 (c2 (6),c)) DEFAULT CHARSET=latin1 ROW_FORMAT=COMPACT;
+INSERT INTO t (c) VALUES (1),(2),(3),(4),(5);
+UPDATE t SET c2=NULL WHERE c = 1;
+UPDATE t SET c2="" WHERE c=1;
+UPDATE t SET c2="TESTtest" WHERE c = 1;
+UPDATE t SET c2="BUGbug" WHERE c2 = "TESTtest";
+DELETE FROM t WHERE c2="BUGbug";
+SELECT * FROM t;
+DROP TABLE t;
+CREATE TABLE t (c INT,c2 TEXT,KEY k2 (c2 (6),c)) DEFAULT CHARSET=latin1 ROW_FORMAT=COMPACT;
+INSERT INTO t (c) VALUES (1),(1),(1),(1),(1);
+UPDATE t SET c2=NULL WHERE c = 1;
+UPDATE t SET c2="TESTtest" WHERE c = 1;
+UPDATE t SET c2="BUGbug" WHERE c2 = "TESTtest";
+DELETE FROM t WHERE c2="BUGbug";
+SELECT * FROM t;
+DROP TABLE t;
+CREATE TABLE t (c INT,c2 TEXT NOT NULL,KEY k2 (c2 (6),c)) DEFAULT CHARSET=latin1 ROW_FORMAT=COMPACT;
+INSERT INTO t (c) VALUES (1),(1),(1),(1),(1);
+UPDATE t SET c2=NULL WHERE c = 1;
+UPDATE t SET c2="TESTtest" WHERE c = 1;
+UPDATE t SET c2="BUGbug" WHERE c2 = "TESTtest";
+DELETE FROM t WHERE c2="BUGbug";
+SELECT * FROM t;
+DROP TABLE t;
+CREATE TABLE t (c INT NOT NULL PRIMARY KEY,c2 TEXT,KEY k2 (c2 (6),c)) DEFAULT CHARSET=latin1 ROW_FORMAT=COMPACT;
+INSERT INTO t (c) VALUES (1),(2),(3),(4),(5);
```

Comment:
this test now has column 'c' as primary key, and wsrep patch will then append only key for the 'c' column into the write set. The original problem statement says: 
Problem was that row_mysql_read_blob_ref can return NULL in case when blob datatype is used in a key and its real value is NULL.
To stress this scenario, the table should have a unique key, where a blob column is one part of the key, but has NULL value.  Changing the KEY k2 to UNIQUE KEY k2 would yield this. 
Note that wsrep patch appends key values only for primary keys, unique keys and foreign keys into the write set

Reply:
Added.